### ISSUE TITLE
Deno logs "BadResource: Bad resource ID" error running NextJS dev task

### ISSUE THREAD

**Issue by barberdt** [denoland/deno#26343]
Version: Deno 2.0.1

After upgrading to Deno 2.0.1, I started observing an error being reported when making requests to my NextJS app served using `deno task dev`. Reverting to 2.0.0 mitigates this error, which makes me suspect some kind of regression.

### Repro steps (using asdf to manage Deno version 2.0.1)

```zsh
➜  ~  deno run -A npm:create-next-app@latest // use all default settings
➜  ~  cd deno-bug
➜  deno-bug git:(main) ✗ asdf local deno 2.0.1
➜  deno-bug git:(main) ✗ deno task dev
Task dev next dev
  ▲ Next.js 14.2.15
  - Local:        http://localhost:3000

 ✓ Starting...
 ✓ Ready in 962ms
 ○ Compiling / ...
 ✓ Compiled / in 637ms (528 modules)
 GET / 200 in 760ms
 ✓ Compiled in 90ms (254 modules)
Error handling upgrade request BadResource: Bad resource ID
    at TcpConn.setNoDelay (ext:deno_net/01_net.js:211:12)
    at TCP.setNoDelay (ext:deno_node/internal_binding/tcp_wrap.ts:229:28)
    at Socket.setNoDelay (node:net:595:20)
    at WebSocket.setSocket (file:///Users/davis/barberdt/deno-bug/node_modules/next/dist/compiled/ws/index.js:1:34670)
    at WebSocketServer.completeUpgrade (file:///Users/davis/barberdt/deno-bug/node_modules/next/dist/compiled/ws/index.js:1:31598)
    at WebSocketServer.handleUpgrade (file:///Users/davis/barberdt/deno-bug/node_modules/next/dist/compiled/ws/index.js:1:30694)
    at HotReloaderWebpack.onHMR (file:///Users/davis/barberdt/deno-bug/node_modules/next/dist/server/dev/hot-reloader-webpack.js:349:18)
    at upgradeHandler (file:///Users/davis/barberdt/deno-bug/node_modules/next/dist/server/lib/router-server.js:459:59)
    at ServerImpl.<anonymous> (file:///Users/davis/barberdt/deno-bug/node_modules/next/dist/server/lib/start-server.js:170:19)
    at ServerImpl.emit (ext:deno_node/_events.mjs:393:28)
    at handler (node:http:1310:14)
    at ext:deno_http/00_serve.ts:369:24
    at ext:deno_http/00_serve.ts:577:29
    at eventLoopTick (ext:core/01_core.js:175:7) {
  name: "BadResource"
}
 ✓ Compiled /favicon.ico in 83ms (305 modules)
 GET /favicon.ico 200 in 120ms
```

### Success steps (using asdf to manage Deno version 2.0.0)

```zsh
➜  ~  deno run -A npm:create-next-app@latest // use all default settings
➜  ~  cd deno-bug
➜  deno-bug git:(main) ✗ asdf local deno 2.0.0
➜  deno-bug git:(main) ✗ deno task dev
Task dev next dev
  ▲ Next.js 14.2.15
  - Local:        http://localhost:3000

 ✓ Starting...
 ✓ Ready in 960ms
 ○ Compiling / ...
 ✓ Compiled / in 625ms (528 modules)
 GET / 200 in 728ms
 ✓ Compiled in 93ms (254 modules)
 ✓ Compiled /favicon.ico in 81ms (305 modules)
 GET /favicon.ico 200 in 116ms
```

**Comment by nathanwhit** [comment]
Likely fixed by #26342

**Comment by nathanwhit** [comment]
Fix landed in 2.0.2
